bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/PWAImpl.java

```diff
@@ -48,26 +48,31 @@
 
     @PostConstruct
     protected void initModel() {
-        ValueMap valueMap = resource.getValueMap();
-        Boolean isPWAEnabled = valueMap.get(PN_PWA_ENABLED, Boolean.class);
-        this.isEnabled = (isPWAEnabled != null) ? isPWAEnabled : false;
-        if (!this.isEnabled) {
+        PageManager pageManager = resource.getResourceResolver().adaptTo(PageManager.class);
+        if (pageManager == null) {
             return;
         }
 
-        this.themeColor = colorToHex(valueMap.get(PN_PWA_THEME_COLOR, ""));
-        this.iconPath = valueMap.get(PN_PWA_ICON_PATH, "");
-
-        String startURL = valueMap.get(PN_PWA_START_URL, "");
-        this.manifestPath = replaceSuffix(startURL, MANIFEST_NAME);
-
-        PageManager pageManager = resource.getResourceResolver().adaptTo(PageManager.class);
-        if (pageManager != null) {
-            Page page = pageManager.getContainingPage(resource);
-            if (page != null) {
+        Page page = pageManager.getContainingPage(resource);
+        while (page != null) {
+            Resource contentResource = page.getContentResource();
+            if (contentResource == null) {
```

Comment:
I find the logic here (L59-75, using `continue` and `break`) unnecessary complicated. I would write it as:
```java
if (contentResource != null) {
    ValueMap valueMap = contentResource.getValueMap();
    Boolean isPWAEnabled = valueMap.get(PN_PWA_ENABLED, Boolean.class);
    if (isPWAEnabled != null && isPWAEnabled) {
        ...
        return; //could be break if you need more operations after the loop
    }
}
page = page.getParent();
```